x-pack/packages/observability/observability_utils/observability_utils_server/entities/signals/get_alerts_for_entity.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { RulesClient } from '@kbn/alerting-plugin/server';
+import { AlertsClient } from '@kbn/rule-registry-plugin/server';
+import {
+  ALERT_GROUP_FIELD,
+  ALERT_GROUP_VALUE,
+  ALERT_STATUS,
+  ALERT_STATUS_ACTIVE,
+  ALERT_TIME_RANGE,
+} from '@kbn/rule-data-utils';
+import { kqlQuery } from '../../es/queries/kql_query';
+import { rangeQuery } from '../../es/queries/range_query';
+
+export async function getAlertsForEntity({
+  start,
+  end,
+  entity,
+  alertsClient,
+  rulesClient,
+  size,
+}: {
+  start: number;
+  end: number;
+  entity: Record<string, unknown>;
+  alertsClient: AlertsClient;
+  rulesClient: RulesClient;
+  size: number;
+}) {
+  const alertsKuery = Object.entries(entity)
+    .map(([field, value]) => {
+      return `(${ALERT_GROUP_FIELD}:"${field}" AND ${ALERT_GROUP_VALUE}:"${value}")`;
```

Comment:
I believe some of the rules are missing adding entity fields, like `service.name`, under `kibana.alert.group.field` and `kibana.alert.group.value`, but do have the ecs fields like `service.name` in the AAD.

Reply:
thanks!